main.go

```diff
@@ -1,54 +1,45 @@
 package main
 
 import (
-	"fmt"
+	"os"
+	"os/signal"
 	"rest-gateway/conf"
+	"rest-gateway/handlers"
 	"rest-gateway/logger"
 	"rest-gateway/router"
+	"syscall"
 	"time"
-
-	"github.com/memphisdev/memphis.go"
 )
 
-func main() {
+func initalizeLogger() {
 	configuration := conf.GetConfig()
-	var conn *memphis.Conn
 	ticker := time.NewTicker(1 * time.Second)
 	for {
 		select {
 		case <-ticker.C:
-			var err error
-			opts := []memphis.Option{memphis.Reconnect(true), memphis.MaxReconnect(10), memphis.ReconnectInterval(3 * time.Second)}
+			creds := configuration.CONNECTION_TOKEN
+			username := configuration.ROOT_USER
 			if configuration.USER_PASS_BASED_AUTH {
-				opts = append(opts, memphis.Password(configuration.ROOT_PASSWORD))
-			} else {
-				opts = append(opts, memphis.ConnectionToken(configuration.CONNECTION_TOKEN))
-			}
-			if configuration.CLIENT_CERT_PATH != "" && configuration.CLIENT_KEY_PATH != "" && configuration.ROOT_CA_PATH != "" {
-				opts = append(opts, memphis.Tls(configuration.CLIENT_CERT_PATH, configuration.CLIENT_KEY_PATH, configuration.ROOT_CA_PATH))
+				username = "$memphis"
+				creds = configuration.CONNECTION_TOKEN + "_" + configuration.ROOT_PASSWORD
 			}
-			conn, err = memphis.Connect(configuration.MEMPHIS_HOST, configuration.ROOT_USER, opts...)
-			if err == nil {
-				ticker.Stop()
-				goto serverInit
-			} else {
-				fmt.Printf("Awaiting to establish connection with Memphis - %v\n", err.Error())
+			l, err := logger.CreateLogger(configuration.MEMPHIS_HOST, username, creds)
+			if err != nil {
+				panic("Logger creation failed - " + err.Error())
```

Comment:
The main purpose of this ticker for loop is to retry in case that connection didn't succeed, if you are panic it stops after 1 try
in this case you should just print
fmt.Printf("Awaiting to establish connection with Memphis - %v\n", err.Error())